### PR TITLE
feat: add prefer-async-await rule

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[X] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### CODE REVIEW

**Comment by incendial** [comment]
@roman-petrov could you please take a look too? Maybe I missed some cases.

**Comment by roman-petrov** [comment]
@incendial this looks promising! I am a bit ill and surely will take closer look in a couple of days.

**Comment by incendial** [comment]
> @incendial this looks promising! I am a bit ill and surely will take closer look in a couple of days.

Sure, get well soon!

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

File: lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_async_await/prefer_async_await_rule.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    source.unit.visitChildren(visitor);
+
+    return visitor.invocations
+        .map((invocation) => createIssue(
+              rule: this,
+              location: nodeLocation(
+                node: invocation,
+                source: source,
+                withCommentOrMetadata: true,
```
[the remark this answers is not in the document]
Comment (incendial) [reply lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_async_await/prefer_async_await_rule.dart]: Fixed

**Review by dkrutskikh — APPROVED** [review]

**Pull request merged**